### PR TITLE
Update metafiles and license

### PR DESCRIPTION
**Pull request by marcofavorito**
- update LICENSE (from GPLv3+ to LGPLv3+)
- update README (license section)
- update HISTORY with release 0.1.0 details
- refactoring of some parts of the Dockerfile (e.g. dependency versions
  as environment variable instead of hard-coded)
- set 'CMAKE_SKIP_INSTALL_ALL_DEPENDENCY' such that 'make install' won't
  build other targets like test and benchmark
- add arg to Dockerfile to accept a git reference to the Lydia repository to be compiled.

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] I have read the **CONTRIBUTING** document.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -86,7 +86,7 @@ set(CMAKE_MODULE_PATH
 
 set( CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin )
 set( CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib )
-
+set (CMAKE_SKIP_INSTALL_ALL_DEPENDENCY TRUE)
```
Comment (marcofavorito) [inline CMakeLists.txt:89]: I will leave a comment to the relevant changes, as the majority is just due to the license replacement.

File: Dockerfile [hunk 2]
```diff
@@ -40,50 +40,61 @@ RUN HOME=/home/default && \
     usermod -a -G sudo default && \
```
Comment (marcofavorito) [inline Dockerfile:40]: check the `Dockerfile`

File: HISTORY.md [hunk 3]
```diff
@@ -1 +1,14 @@
-# History
```
Comment (marcofavorito) [inline HISTORY.md:1]: check the History

File: LICENSE [hunk 4]
```diff
@@ -1,674 +1,165 @@
-                    GNU GENERAL PUBLIC LICENSE
```
Comment (marcofavorito) [inline LICENSE:1]: new license, LGPL

File: README.md [hunk 5]
```diff
@@ -56,7 +56,9 @@ lydia -l ltlf -f examples/counter_1.ltlf --part examples/counter_1.part
 ```
```
Comment (marcofavorito) [inline README.md:56]: check the README

Comment (francescofuggitti) [reply README.md:56]: what about adding acknowledgements to Whitemech and Tailor?

File: Dockerfile (hunk 2 above)
Comment (francescofuggitti) [reply Dockerfile:40]: I get two errors:

1. `./scripts/docker-build.sh: line 3: ${$1:-develop}: bad substitution`
2. Missing `CUDD_INCLUDE_DIRS` in Lydia configuration, `Could NOT find CUDD (missing: CUDD_INCLUDE_DIRS)`

Comment (marcofavorito) [reply Dockerfile:40]: thanks! fixed with 
4cc0d43

**Review by francescofuggitti — APPROVED** [review]
LGTM

File: README.md (hunk 5 above)
Comment (marcofavorito) [reply README.md:56]: right, will do :-)

File: README.md [hunk 6]
```diff
@@ -255,10 +257,20 @@ Hence, you have to make sure new versions of `flex` and `bison` executables are
...
+
+## License
+
+This software is released under the GNU Lesser General Public License version 3 or later.
+
+## Acknowledgements
+
+This work has been funded by the [ERC Advanced Grant "WhiteMech"](whitemech.github.io/)
+and by the [TAILOR research network](https://tailor-network.eu/).
+
```
Comment (marcofavorito) [inline README.md:276]: @francescofuggitti please let me know your thoughts, and please feel free to change as you see fit!

Comment (francescofuggitti) [reply README.md:276]: I would say "has been partially funded". Then, we could be precise with grants numbers, but I think it is also ok as is

File: README.md [hunk 7]
```diff
@@ -255,10 +257,20 @@ Hence, you have to make sure new versions of `flex` and `bison` executables are
...
 ```
+
+## License
+
+This software is released under the GNU Lesser General Public License version 3 or later.
+
+## Acknowledgements
+
+This work has been funded by the [ERC Advanced Grant "WhiteMech"](whitemech.github.io/)
+and by the [TAILOR research network](https://tailor-network.eu/).
```
Comment (francescofuggitti) [inline README.md:275]:
```suggestion
This work has been partially funded by the [ERC Advanced Grant "WhiteMech"](whitemech.github.io/)
and by the [TAILOR research network](https://tailor-network.eu/).
```

**Pull request merged**